src/renderers/shared/fiber/ReactFiberClassComponent.js

```diff
@@ -96,17 +97,89 @@ module.exports = function(scheduleUpdate : (fiber: Fiber, priorityLevel : Priori
     return true;
   }
 
+  function checkClassInstance(workInProgress: Fiber, inst: any) {
+    const type = workInProgress.type;
+    const constructor = inst && inst.constructor;
+    const name = (
+      type.displayName || (constructor && constructor.displayName) ||
+      type.name || (constructor && constructor.name) ||
+      'A Component'
+    );
+    if (__DEV__) {
+      warning(
+        inst.render,
+        '%s(...): No `render` method found on the returned component ' +
+        'instance: you may have forgotten to define `render`.',
+        name
+      );
+      warning(
+        !inst.getInitialState ||
+        inst.getInitialState.isReactClassApproved,
+        'getInitialState was defined on %s, a plain JavaScript class. ' +
+        'This is only supported for classes created using React.createClass. ' +
+        'Did you mean to define a state property instead?',
+        name
+      );
+      warning(
+        !inst.getDefaultProps ||
+        inst.getDefaultProps.isReactClassApproved,
+        'getDefaultProps was defined on %s, a plain JavaScript class. ' +
+        'This is only supported for classes created using React.createClass. ' +
+        'Use a static property to define defaultProps instead.',
+        name
+      );
+      ['propTypes', 'contextTypes'].forEach(instProp => {
+        warning(
+          !inst[instProp],
+          '%s was defined as an instance property on %s. Use a static ' +
+          'property to define %s instead.',
+          instProp,
+          name,
+          instProp
+        );
+      });
+      warning(
+        typeof inst.componentShouldUpdate !== 'function',
+        '%s has a method called ' +
+        'componentShouldUpdate(). Did you mean shouldComponentUpdate()? ' +
+        'The name is phrased as a question because the function is ' +
+        'expected to return a value.',
+        name
+      );
+      warning(
+        typeof inst.componentDidUnmount !== 'function',
+        '%s has a method called ' +
+        'componentDidUnmount(). But there is no such lifecycle method. ' +
+        'Did you mean componentWillUnmount()?',
+        name
+      );
+      warning(
+        typeof inst.componentWillRecieveProps !== 'function',
+        '%s has a method called ' +
+        'componentWillRecieveProps(). Did you mean componentWillReceiveProps()?',
+        name
+      );
+    }
+
+    if (inst.state && (typeof inst.state !== 'object' || Array.isArray(inst.state))) {
+      throw new Error(`${name}.state: must be set to an object or null`);
```

Comment:
Can you add a comment like `// TODO: Change this to be a warning.`?

I don't think this needs to be an error but might as well do this for compatibility for now.